modules/openapi-generator-api/src/main/java/org/openapitools/codegen/api/TemplatingGenerator.java

```diff
@@ -0,0 +1,11 @@
+package org.openapitools.codegen.api;
+
+/*
+interface for getting the templates
+ */
+public interface TemplatingGenerator {
+
+  // returns the template content by name
+  String getFullTemplate(String name);
```

Comment:
This name isn't too clear to me (is it a full path to the template, or template contents)? The comment clarifies this, and should be a standard JavaDoc comment.

Could this be renamed to `getFullTemplateContents` for clarity? Usually, naming isn't that important, but this will quickly become a method name required by implementers which makes it harder to change later.

Reply:
OK, getFullTemplateContents seems better to me also :-)